include/moqt/core/transport.h

```diff
@@ -0,0 +1,226 @@
+
+        enum class Status : uint8_t
+        {
+            kReady = 0,
+            kNotReady,
+
+            kInternalError,
+
+            kInvalidParams,
+
+            kClientConnecting,
+            kClientNotConnected,
+            kClientFailedToConnect
+        };
+
+        /**
+         * @brief Client mode Constructor to create the MOQ instance
+         *
+         * @param cfg       MoQT Instance Client Configuration
+         */
+        Transport(const ClientConfig& cfg);
+
+        /**
+         * @brief Server mode Constructor to create the MOQ instance
+         *
+         * @param cfg        MoQT Server Configuration
+         */
+        Transport(const ServerConfig& cfg);
+
+        ~Transport() = default;
+
+        // -------------------------------------------------------------------------------------------------
+        // Public API MoQ Intance API methods
+        // -------------------------------------------------------------------------------------------------
+        /**
+         * @brief Subscribe to a track
+         *
+         * @param conn_id           Connection ID to send subscribe
+         * @param track_delegate    Track delegate to use for track related functions and callbacks
+         *
+         * @returns `track_alias` if no error and nullopt on error
+         */
+        std::optional<uint64_t> SubscribeTrack(TransportConnId conn_id,
+                                               std::shared_ptr<SubscribeTrackHandler> track_delegate);
+
+        /**
+         * @brief Unsubscribe track
+         *
+         * @param conn_id           Connection ID to send subscribe
+         * @param track_delegate    Track delegate to use for track related functions and callbacks
+         */
+        void UnsubscribeTrack(TransportConnId conn_id, std::shared_ptr<SubscribeTrackHandler> track_delegate);
+
+        /**
+         * @brief Publish to a track
+         *
+         * @param conn_id           Connection ID from transport for the QUIC connection context
+         * @param track_delegate    Track delegate to use for track related functions
+         *                          and callbacks
+         *
+         * @returns `track_alias` if no error and nullopt on error
+         */
+        std::optional<uint64_t> PublishTrack(TransportConnId conn_id,
+                                             std::shared_ptr<PublishTrackHandler> track_delegate);
+
+        /**
+         * @brief Unpublish track
+         *
+         * @param conn_id           Connection ID from transport for the QUIC connection context
+         * @param track_delegate    Track delegate used when published track
+         */
+        void UnpublishTrack(TransportConnId conn_id, std::shared_ptr<PublishTrackHandler> track_delegate);
+
+        /**
+         * @brief Get the instance status
+         *
+         * @return Status indicating the state/status of the instance
+         */
+        Status GetStatus();
```

Comment:
There is Status and TransportStatus within Transport. Wondering if we need to merge them or rename it to better reflect the context

Reply:
This is the MOQ Transport, which is integrates the underlining QUIC transport. From an API standpoint, this API does not directly interact with the QUIC transport, all that is provided within MoQT API.  A topic for discussion next week is around moving qtransport into libquicr.